### PR TITLE
[HAPI-594 OF-136] Extract IGB client from PKB for both HAPI JP and Compliance use

### PR DESCRIPTION
**Pull request by marianmazarovici**

### CODE REVIEW

File: igb_client/client.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    _base_url = "{base_url}/{environment_id}/{view}"
+    _credentials_storage_key = None
+
+    def __init__(self, api_key: str, environment_id: str, credentials_storage_key: str,
+                 base_url="https://api.ingoedebanen.nl/apipartner/hapi/v1"):
+        self._environment_id = environment_id
+        self._credentials_storage_key = credentials_storage_key
+        self._base_url = self._base_url.format(base_url=base_url, environment_id=environment_id)
+
+        self.session = requests_cache.CachedSession('hapi_ofccp_cache')
```
Comment (gabalese) [inline igb_client/client.py]: - Probably better to use a more general name than `hapi_ofccp`
- Worth specifing an `expire_after` parameter here.

File: igb_client/client.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            cls._instance = super().__new__(cls)
+            cls._instance.__init__(api_key, environment_id, credentials_storage_key)
+        return cls._instance
+
+    def encrypt_credentials(self, credentials: Credential) -> Credential:
+        credentials.credentials = {
+            k: AESCypher(self._credentials_storage_key).encrypt(v)
+            for k, v in credentials.credentials.items()
+        }
+        return credentials
```
Comment (gabalese) [inline igb_client/client.py:45]: Is this missing a method to decrypt those credentials?

File: igb_client/client.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+class IGBClientBase:
+    _instance = None
+    _base_url = "{base_url}/{environment_id}/{view}"
+    _credentials_storage_key = None
+
+    def __init__(self, api_key: str, environment_id: str, credentials_storage_key: str,
+                 base_url="https://api.ingoedebanen.nl/apipartner/hapi/v1"):
+        self._environment_id = environment_id
+        self._credentials_storage_key = credentials_storage_key
```
Comment (gabalese) [inline igb_client/client.py:23]: What about the credentials _transport_ key?

Comment (marianmazarovici) [reply igb_client/client.py:23]: added key and helper methods

File: igb_client/client.py (hunk 2 above)
Comment (marianmazarovici) [reply igb_client/client.py:45]: added helper method

File: igb_client/client.py (hunk 1 above)
Comment (marianmazarovici) [reply igb_client/client.py]: fwded expire_after parameter and gave the session a more meaningful name

**Review by gabalese — APPROVED** [review]

**Pull request merged**